src/exchange/billing/account-renew/billing-account-renew.controller.js

```diff
@@ -0,0 +1,256 @@
+    }
+
+    const accountInChange = _(this.buffer.changes).find({
+      primaryEmailAddress: account.primaryEmailAddress,
+    });
+
+    if (accountInChange == null) {
+      this.buffer.changes.push({
+        primaryEmailAddress: account.primaryEmailAddress,
+        primaryEmailDisplayName: account.primaryEmailDisplayName,
+        renewPeriod: account.renewPeriod,
+        exchangeOffer: this.exchange.offer,
+      });
+    } else {
+      accountInChange.renewPeriod = account.renewPeriod;
+    }
+  }
+
+  retrieveAccounts(count, offset) {
+    this.loading = true;
+
+    this.services.Exchange
+      .getAccountsForExchange(this.exchange, count, offset, this.search.value)
+      .then((accounts) => {
+        this.accounts = accounts;
+        this.bufferedAccounts = _.cloneDeep(accounts);
+
+        if (
+          _(this.bufferedAccounts).has('list.results')
+            && this.bufferedAccounts.list.results != null
+        ) {
+          this.buffer.ids = this.bufferedAccounts.list.results.map(
+            item => item.primaryEmailAddress,
+          );
+
+          // roll previous buffered changes
+          if (this.buffer.hasChanged) {
+            _.forEach(this.bufferedAccounts.list.results, (currentBufferedAccount) => {
+              const buffer = _(this.buffer.changes).find({
+                primaryEmailAddress: currentBufferedAccount.primaryEmailAddress,
+              });
+
+              if (buffer != null) {
+                _.set(currentBufferedAccount, 'renewPeriod', buffer.renewPeriod);
+              }
+            });
+          }
+
+          // needed by selectAll checkbox
+          _.forEach(this.bufferedAccounts.list.results, (account) => {
+            this.trackSelected(account.primaryEmailAddress, account.renewPeriod);
+          });
+        }
+      })
+      .catch((failure) => {
+        this.onError({
+          result: `${this.services.$translate.instant('exchange_tab_ACCOUNTS_error_message')} ${failure}`,
+        });
+      })
+      .finally(() => {
+        this.loading = false;
+      });
+  }
+
+  /**
+     * Mark alltems on the page as selected with 'value'.
+     * @param value
+     */
+  checkboxStateChange(value) {
+    if (_.has(this.buffer, 'ids') && this.buffer.ids != null) {
+      _.forEach(this.buffer.ids, (id) => {
+        this.trackSelected(id, value);
+      });
+    }
+  }
+
+  trackSelected(primaryEmailAddress, period) {
+    const matchingAccount = _(this.bufferedAccounts.list.results).find({
+      primaryEmailAddress,
+    });
+
+    if (period !== this.buffer.periodSelectedForAll) {
+      this.buffer.periodSelectedForAll = null;
+    }
+
+    if (matchingAccount != null) {
+      matchingAccount.renewPeriod = period;
+
+      this.checkForChanges();
+    }
+
+    if (this.bufferedAccounts.list.results.every(({ renewPeriod }) => renewPeriod === period)) {
+      this.buffer.periodSelectedForAll = period;
+    }
+  }
+
+  static GetPropertyNameFromPeriodName(period) {
```

Comment:
static method should be placed at bottom of file

Reply:
Is there any reason of doing this?

I'm not sure that ordered methods in that way helps.